### PR TITLE
Fix crash in publication page and add error boundaries

### PR DESCRIPTION
**Pull request by philippotto**
- adds error boundaries to the individual routes (except for the tracing views since these have their own boundaries)
  - a "nice" error page will be shown to the user and the error will be logged to airbrake, too
- adds top-level error boundary for route-independent react errors
- fixes rendering error when a publication doesn't have any active datasets or annotations

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- the publication fix is a bit tedious to set up. I don't think, it needs additional testing. It will look like this (a bit naked, but this should be only an edge case, anyway):
image
- the error boundaries can be tested by:
  - navigating to admin > users
  - image
  - clicking "help" in the navbar and "ask a question" (this error boundary is route independent)
image

### Issues:
- see https://scm.slack.com/archives/C5AKLAV0B/p1671099320485249 for context

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/admin/user/user_list_view.tsx [hunk 1]
```diff
@@ -288,6 +288,7 @@ class UserListView extends React.PureComponent<Props, State> {
   };
 
   render() {
+    throw new Error("Dummy error to test error boundary");
```
Comment (philippotto) [inline frontend/javascripts/admin/user/user_list_view.tsx:291]:
```suggestion
```

remove after testing/approval.

File: frontend/javascripts/navbar.tsx [hunk 2]
```diff
@@ -316,6 +316,10 @@ function HelpSubMenu({
 } & SubMenuProps) {
   const [isHelpModalOpen, setIsHelpModalOpen] = useState(false);
 
+  if (isHelpModalOpen) {
+    throw new Error("Throwing dummy error because you clicked on Help > Ask a question.");
+  }
+
```
Comment (philippotto) [inline frontend/javascripts/navbar.tsx:322]:
```suggestion
```

remove after testing/approval.

**Review by philippotto — COMMENTED** [review]
review will be easier by ignoring whitespace changes

**Review by normanrz — APPROVED** [review]
code looks good. i'll do some testing now

**Comment by normanrz** [comment]
Tested -> works. I was wondering if the toast is still required? Seems a bit redundant.

**Comment by philippotto** [comment]
> Tested -> works. I was wondering if the toast is still required? Seems a bit redundant.

I assume you tested it locally, right? Orthogonal to the error boundaries, there is a global `window.onerror` handler (mainly for errors unrelated to react). Due to a peculiarity of react, that event handler is triggered again after an error boundary was activated, but only in DEV mode. See this [react issue](https://github.com/facebook/react/issues/10474). So, in the production build, there won't be the redundant error toast. I just tested it at https://fixpublicationpage.webknossos.xyz/users.
